java/test/com/facebook/watchman/WatchmanConnectionTest.java

```diff
@@ -295,5 +297,83 @@ public void onFailure(Throwable t) {
       entry++;
     }
   }
+
+  @Test
+  public void propagatesDeserializationException() throws Exception {
```

Comment:
Added this unit test to make sure that if we see the BserEofException that's been showing up in stack traces, that the exception will get propagated back to the user of this library.